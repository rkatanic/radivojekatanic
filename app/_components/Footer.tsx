const Footer = () => (
  <footer>
    <div className="mt-20 flex flex-col justify-between gap-4 border-t border-zinc-200 py-12 text-center text-sm text-zinc-500 dark:border-zinc-800 lg:flex-row lg:items-center lg:text-left">
      <div>
        Â© {new Date().getFullYear()} Radivoje Katanic. All rights reserved.
      </div>
      <div>Built with Next.js and simplicity in mind.</div>
    </div>
  </footer>
);

export default Footer;
