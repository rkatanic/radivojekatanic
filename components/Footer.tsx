import getConfig from "next/config";
import { DATE_FORMAT } from "utils/dateTimeUtils";

const Footer = (): JSX.Element => {
  const currentYear = new Date().getFullYear();

  const { publicRuntimeConfig } = getConfig();
  const lastModifiedDate = DATE_FORMAT.format(
    new Date(publicRuntimeConfig.lastModifiedDate)
  );

  return (
    <div className="mt-12 w-full border-t py-6 text-sm">
      <div className="m-auto flex max-w-4xl items-center justify-between text-gray-400">
        <span>Â© {currentYear} Radivoje Katanic</span>
        <span>Last updated on {lastModifiedDate}</span>
      </div>
    </div>
  );
};

export default Footer;
