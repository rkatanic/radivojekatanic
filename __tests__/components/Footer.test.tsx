import Footer from "@/components/Footer";
import { render } from "@testing-library/react";

jest.mock("next/config", () => () => ({
  publicRuntimeConfig: {
    lastModifiedDate: "01-10-2020",
  },
}));

describe("Footer", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Footer />);

    expect(baseElement).toMatchSnapshot();
  });
});
