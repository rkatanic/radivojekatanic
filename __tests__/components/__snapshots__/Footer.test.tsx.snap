// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<body>
  <div>
    <div
      class="mt-12 w-full border-t py-6 text-sm"
    >
      <div
        class="m-auto flex max-w-4xl items-center justify-between text-gray-400"
      >
        <span>
          Â© 
          2023
           Radivoje Katanic
        </span>
        <span>
          Last updated on 
          Jan 10, 2020
        </span>
      </div>
    </div>
  </div>
</body>
`;
